OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,440 @@
+                    {
+                        ____value.y += 1f;
+                    }
+                }
+            }
+        }
+
+        internal static ModInputHandler Self { get; private set; }
+
+        private HashSet<IModInputCombination> _singlePressed = new HashSet<IModInputCombination>();
+        private Dictionary<Int64, IModInputCombination> _comboRegistry = new Dictionary<Int64, IModInputCombination>();
+        private HashSet<InputCommand> _gameBindingRegistry = new HashSet<InputCommand>();
+        private float[] _timeout = new float[_maxUsefulKey];
+        private int[] _gameBindingCounter = new int[_maxUsefulKey];
+        private ModInputCombination _lastPressed;
+        private float _lastUpdate;
+        private float _lastSingleUpdate;
+        private readonly static float _updateCooldown = 0.01f; //aka little bit less than one frame
+        private readonly static float _cooldown = 0.05f;
+        private readonly static float _tapKeep = 0.3f;
+        private readonly static float _tapDuration = 0.1f;
+        private readonly static int _minUsefulKey = 8;
+        private readonly static int _maxUsefulKey = 350;
+        private readonly IModLogger _logger;
+        private readonly IModConsole _console;
+
+        internal bool ShouldIgnore(KeyCode code)
+        {
+            UpdateCombo();
+            return _lastPressed != null && Time.realtimeSinceStartup - _timeout[(int)code] < _cooldown;
+        }
+
+        public ModInputHandler(IModLogger logger, IModConsole console, IHarmonyHelper patcher)
+        {
+            _console = console;
+            _logger = logger;
+            patcher.AddPrefix<SingleAxisCommand>("ClearBinding", typeof(InputInterceptor), nameof(InputInterceptor.SingleAxisRemovePre));
+            patcher.AddPrefix<DoubleAxisCommand>("ClearBinding", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisRemovePre));
+            patcher.AddPostfix<SingleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.SingleAxisUpdatePost));
+            patcher.AddPostfix<DoubleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisUpdatePost));
+            Self = this;
+        }
+
+        private void UpdateCombo()
+        {
+            if (Time.realtimeSinceStartup - _lastUpdate > _updateCooldown)
+            {
+                _lastUpdate = Time.realtimeSinceStartup;
+                Int64 hash = 0;
+                int[] keys = new int[7];
+                int t = 0;
+                bool countdownTrigger = false;
+                for (int code = _minUsefulKey; code < _maxUsefulKey; code++)
+                {
+                    if (Enum.IsDefined(typeof(KeyCode), (KeyCode)code) && Input.GetKey((KeyCode)code))
+                    {
+                        keys[t] = code;
+                        t++;
+                        if (t > 7)
```

Comment:
Is this the same 7 as the one above? Magic numbers :)

Reply:
done, added _maxComboLength constant